plugins/com.gwtplugins.gwt.eclipse.wtp/src/com/google/gwt/eclipse/wtp/utils/GwtFacetUtils.java

```diff
@@ -2,22 +2,97 @@
 
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.wst.common.project.facet.core.FacetedProjectFramework;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.server.core.IModule;
+import org.eclipse.wst.server.core.IServer;
 
-import com.google.gwt.eclipse.wtp.GwtWtpPlugin;
 import com.google.gwt.eclipse.wtp.facet.data.IGwtFacetConstants;
 
 public class GwtFacetUtils {
 
-  public static boolean hasGwtFacet(IProject project) {
-    boolean hasFacet = false;
-    try {
-      hasFacet = FacetedProjectFramework.hasProjectFacet(project, IGwtFacetConstants.GWT_FACET_ID);
-    } catch (CoreException e) {
-      GwtWtpPlugin.logError("hasGetFacet: Error, can't figure GWT facet.", e);
+  /**
+   * Does any of the projects added to the server runtime have a GWT Facet? 
+   * 
+   * @param server wtp runtime server
+   * @return returns true if a GWT facet was found
+   */
+  public static boolean hasGwtFacet(IServer server) {
+    IModule[] modules = server.getChildModules(server.getModules(), new NullProgressMonitor());
+    if (modules == null || modules.length == 0) {
+      return false;
     }
 
-    return hasFacet;
+    for (IModule module : modules) {
```

Comment:
This will fail if there is more than one gwt facet. Not sure how to deal with that yet.